### DISCUSSION TITLE
Comments on the preprint “Practically significant method comparison protocols for machine learning in small molecule drug discovery” DOI: 10.26434/chemrxiv-2024-6dbwv-v2

### DISCUSSION THREAD

**Discussion by p-w-kenny** [polaris-hub/polaris-method-comparison#2] [Feedback on Guidelines]
At the request of @PatWalters I’ve brought over some comments on the preprint “Practically significant method comparison protocols for machine learning in small molecule drug discovery” (DOI: 10.26434/chemrxiv-2024-6dbwv-v2) that I originally made on LinkedIn. 

Potency, lipophilicity solubility, etc are not actually molecular properties and I recommend not using “molecule” as a synonym for “compound” in the article. While this may seem to be nit-picking errors like this weaken the study and are easily fixed. 

My view is that it is essential to be able to objectively account for numbers of adjustable parameters used to model data if claiming statistical rigor.
 
It is certainly important to have sufficient spread in the data although you need to think beyond “dynamic range” (strictly the variance in the response data should be large in comparison with the uncertainty in the measurement). Although I have used the term “dynamic range” myself in this context I have come to the conclusion that it is incorrect to do so.
 
Clustering in the data can cause validation procedures to make optimistic assessments of model quality (this is actually a tough problem but it’s something that you’ll need to address when defining best practices). My view is that many QSAR/ML models that are claimed to be global are actually ensembles of local models.

I consider genuinely categorical data to be rare in drug discovery and I would argue against categorization of continuous data prior to modelling because it throws information away and distorts relationships between data values. Furthermore, it’s not possible to meaningfully compare predictions with measurements for individual compounds.  Out of range data can still be used for validation.

Some assays have non-obvious limit of quantitation. In my AZ days I found that the MDCK permeability had a very clear upper limit of quantitation which in turn imposed a limit on the maximum efflux ratio that could be measured (at first I was really excited because I thought I’d found a diffusion limit but, after thinking about it a bit, realised that it was due to the assay configuration). Another scenario in which there is a non-obvious limit of quantitation is tight binding inhibition (if the enzyme concentration in the assay is 10 nM you’ll still need 5 nM of a picomolar inhibitor to achieve 50% inhibition).  

These two blog posts may be relevant:

https://fbdd-lit.blogspot.com/2024/10/assessment-of-ai-generated-chemical.html
https://fbdd-lit.blogspot.com/2024/09/variability-in-biological-activity.html

**Comment by PatWalters** [comment]
Peter, thanks for your comments. 
 
>Potency, lipophilicity solubility, etc are not actually molecular properties and I recommend not using “molecule” as a synonym for “compound” in the article. While this may seem to be nit-picking errors like this weaken the study and are easily fixed.
 
You’re correct, we will be more precise with our language. 
 
>My view is that it is essential to be able to objectively account for numbers of adjustable parameters used to model data if claiming statistical rigor.
 
Can you provide a reference to support this belief?  I haven’t seen any mention of this in the statistics literature.   It is important to account for overfitting. Cross-validation accounts for this by assessing performance on a hold out fold with each iteration.

> It is certainly important to have sufficient spread in the data although you need to think beyond “dynamic range” (strictly the variance in the response data should be large in comparison with the uncertainty in the measurement). Although I have used the term “dynamic range” myself in this context I have come to the conclusion that it is incorrect to do so.
 
Good point, we will adjust.  Along these lines, we should mention the importance of calculating the maximum possible correlation using the method developed by Brown, Muchmore, and Hajduk.  This provides a better estimate of “modelabilty” than simply looking at the variance. 
https://www.sciencedirect.com/science/article/pii/S1359644609000403
 
>Clustering in the data can cause validation procedures to make optimistic assessments of model quality (this is actually a tough problem but it’s something that you’ll need to address when defining best practices). My view is that many QSAR/ML models that are claimed to be global are actually ensembles of local models.
 
I’m not sure what you mean here.  We only talk about clustering in the context of dataset splitting.  In this case, we ensure that the same cluster is not in the training and test sets.  I fail to see how that would lead to overly optimistic performance estimates.  I agree that global models are collections of local models.  This is why ensemble methods like Random Forrest work.  I think this is a good thing.  Btw, we are working on a paper that provides recommendations on dataset splitting. 
 
>I consider genuinely categorical data to be rare in drug discovery and I would argue against categorization of continuous data prior to modelling because it throws information away and distorts relationships between data values. Furthermore, it’s not possible to meaningfully compare predictions with measurements for individual compounds. Out of range data can still be used for validation.
 
Most of our paper focuses on regression models.  We simply use classification post-hoc to evaluate how a model would impact decision-making. 
 
>Some assays have non-obvious limit of quantitation. In my AZ days I found that the MDCK permeability had a very clear upper limit of quantitation which in turn imposed a limit on the maximum efflux ratio that could be measured (at first I was really excited because I thought I’d found a diffusion limit but, after thinking about it a bit, realised that it was due to the assay configuration). Another scenario in which there is a non-obvious limit of quantitation is tight binding inhibition (if the enzyme concentration in the assay is 10 nM you’ll still need 5 nM of a picomolar inhibitor to achieve 50% inhibition).
 
I agree, but this seems like a data curation rather than a model comparison issue.

**Reply by p-w-kenny** [reply]
Hi Pat @PatWalters and apologies for the slow response. My understanding is that people doing multivariate regression analysis have recognized that you need to account for number of adjustable parameters for a very long time. The Akaike information criterion (AIC) [DOI](http://doi.org/10.1109/TAC.1974.1100705) has been around for half a century and I’m guessing that adjusted R^2 has been around for even longer. I’ve mentioned AIC and adjusted R^2 just to make the point that the need to account for numbers of adjustable parameters was recognized in ‘traditional’ multivariate regression analysis and I’m not suggesting that either metric would necessarily be suitable for comparing ML models that use different numbers of adjustable parameters. I’m aware that there is a belief that cross-validation eliminates the need to account for the number of adjustable parameters although a quick reading of your recent blog[ post](https://practicalcheminformatics.blogspot.com/2024/11/some-thoughts-on-splitting-chemical.html) left me with the impression that there’s no clear consensus as to how chemical datasets should be split for cross-validation. 

I should explain what I mean by clustering in the training set space because I’m not referring to scaffolds or chemical similarity (that said, ‘survival’ of a model after entire substructural series is held out would greatly increase my confidence in the model and Hawkins, The Problem of Overfitting, JCICS 2004, 44:1-12 [DOI](https://doi.org/10.1021/ci0342472) might be of interest). If a data point in the training set has near neighbours in the training set space (as opposed to neighbours defined in terms of chemical similarity or scaffolds) then these neighbours can, for want of a better term, ‘anchor’ the model when the data point is held out. As an aside, it can be difficult to assess the distribution of data points (or indeed to even define neighbourhoods) within the training set space when the dimensions of the space are correlated.

There are a couple of things that I think need to be considered when ‘global’ models are ensembles of local models. First, the local model for a structural series of interest might actually outperform the global model for the structural series of interest (generally, I would not mix structural series if attempting to build QSAR/ML models for binding affinity or potency). Second, predictions for compounds for which none of the local models are applicable may well be much worse than what you’d expect from cross-validation of the ‘global’ model. My view is that one needs to assess how well compounds fit into the applicability domain of the model when making predictions.

I suggest taking a look at published best practices for QSAR and I believe Alex Tropsha and Paola Grammatica have published on this topic (I’m not suggesting that publications like these are necessarily any good but demonstrating awareness of them will strengthen your own study).

**Comment by cwognum** [comment]
Thanks for the detailed feedback, @p-w-kenny ! And thanks for the detailed response, @PatWalters . 

I'll just chime in here with one other comment. 

> > Clustering in the data can cause validation procedures to make optimistic assessments of model quality (this is actually a tough problem but it’s something that you’ll need to address when defining best practices). My view is that many QSAR/ML models that are claimed to be global are actually ensembles of local models.
>
> I’m not sure what you mean here.

I interpreted this to mean that the dataset itself includes clusters (e.g. different chemotypes). By only reporting a single, global performance metric, you may compare models based on how they do on the largest cluster (i.e. the majority chemotype) rather than across clusters. I think we touch upon this in our paper, but only very briefly (see Section 3.3.4). This is for example discussed in [this blogpost from Inductive Bio](https://www.inductive.bio/blog/building-better-benchmarks-for-adme-optimization) by @benb111 and team. 

This figure from that blog post is great: 
![image](https://github.com/user-attachments/assets/8f4c9bda-e464-4208-97a8-aee6c881c483)

**Reply by p-w-kenny** [reply]
Hi Cas @cwognum and apologies for my slow response.  Hopefully my response to Pat will explain why I think that clustering in the training set space can cause validation procedures to make optimistic assessments of model quality (if not, I’m happy to discuss further). While clusters in the training set space may reflect the presence of structural series, clusters have the potential to lead you astray even they don’t correspond to different structural series. 

The figure that you’ve shown raises a related point. If mean values of the response variable differ between groups of compounds then you’ll achieve a degree of predictivity simply by correctly assigning compound to the appropriate groups. While this does not itself invalidate the model one should anticipate worse predictions when compounds for which the prediction is being made cannot be assigned to one of the groups.

**Comment by cwognum** [comment]
@p-w-kenny One clarifying questions. You say: 

> Potency, lipophilicity solubility, etc are not actually molecular properties

What would you suggest we use instead? Or do you mean to say they are not _molecular_ properties, but rather properties of the compound?

**Reply by p-w-kenny** [reply]
@cwognum I don't think that there is a single term that covers all the properties (which are all quantified as assay results) for which you are aiming to build predictive models and it's generally safe to to consider these to be properties of compounds. I generally use the term 'physicochemical property' for lipophilicity, solubility, passive permeability and melting point. However, solubility of solids generally varies with the nature of the solid state (polymorphs, amorphous solids) and you need to be aware that the solubility that you measure may be limited by salt form (we generally assume that solubility of the neutral form limits the measured value) in which case the nature and concentration of the counterion are relevant. I think that results of assays that assess effects of compounds on targets (and anti-targets) can generally be described as potency (IC50, EC50) or affinity (Kd, Ki) although agonists and activators add complexity. In vitro metabolic stability can be described as a pharmacokinetic property. Are you also planning to build models for in vivo end points including toxicity?

One option would be to create a table in which you group together each class of property (affinity/potency, physicochemical, pharmacokinetic).
